library/core/src/intrinsics/fallback.rs

```diff
@@ -0,0 +1,111 @@
+#![unstable(
+    feature = "core_intrinsics_fallbacks",
+    reason = "The fallbacks will never be stable, as they exist only to be called \
+              by the fallback MIR, but they're exported so they can be tested on \
+              platforms where the fallback MIR isn't actually used",
+    issue = "none"
+)]
+#![allow(missing_docs)]
+
+use crate::panicking::panic_nounwind;
+
+/// Ideally we'd do fallbacks using ordinary trait impls, but that doesn't work
+/// for const (yet™) so we're stuck with hacky workarounds.
+#[inline]
+const fn try_as<T: 'static, F: Copy + 'static>(val: F) -> Option<T> {
+    if const { super::type_id::<T>() == super::type_id::<F>() } {
+        // SAFETY: just checked it's the same type
+        Some(unsafe { super::transmute_unchecked(val) })
+    } else {
+        None
+    }
+}
+
+macro_rules! if_the_types_work {
+    ($f:ident ( $a:expr )) => {
+        if let Some(arg) = try_as($a) {
+            if let Some(ret) = try_as($f(arg)) {
+                return ret;
+            }
+        }
+    };
+}
+
+#[rustc_const_unstable(feature = "core_intrinsics_fallbacks", issue = "none")]
+const fn wide_mul_u128(a: u128, b: u128) -> (u128, u128) {
+    const fn to_low_high(x: u128) -> [u64; 2] {
+        [x as u64, (x >> 64) as u64]
+    }
+    const fn from_low_high(x: [u64; 2]) -> u128 {
+        (x[0] as u128) | ((x[1] as u128) << 64)
+    }
+    #[rustc_const_unstable(feature = "core_intrinsics_fallbacks", issue = "none")]
```

Comment:
If it compiles, it should be fine. ;)

But oh my this is hacky.^^

Reply:
@compiler-errors is currently working on some very unstable const trait support. This can't be used in stable coe any time soon, but if that's an acceptable constraint it may be worth playing around with that?

Reply:
I think it's already turned on in libcore and should be available without bootstrap shenanigans